src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -509,11 +521,15 @@ private Object fetchField(GraphQLFieldDefinition fieldDef, ExecutionContext exec
             CompletableFuture<Object> fetchedValue = (CompletableFuture<Object>) fetchedObject;
             return fetchedValue
                     .handle((result, exception) -> {
+                        executionContext.running();
                         fetchCtx.onCompleted(result, exception);
                         if (exception != null) {
-                            return handleFetchingException(dataFetchingEnvironment.get(), parameters, exception);
+                            CompletableFuture<Object> handleFetchingExceptionResult = handleFetchingException(dataFetchingEnvironment.get(), parameters, exception);
+                            executionContext.finished();
+                            return handleFetchingExceptionResult;
                         } else {
                             // we can simply return the fetched value CF and avoid a allocation
+                            executionContext.finished();
                             return fetchedValue;
```

Comment:
imagine a helper ethod


```
return finished(executionContext, fetchedValue)
```

where this helper method returns `<T>`

I think its neater than trying to out a line of code before each return line